File: Machine/ClientApp/src/ViewModel/coinView.ts
import {CoinCategory} from "../Enum/CoinCategory";

export class CoinView{
	public id: number;
	public coinCategory: CoinCategory;
	public maxCoinCount: number;
	public nowCoinCount: number;
	constructor(id: number, coinCategory: CoinCategory, maxCoinCount: number, nowCoinCount: number) {
		this.id = id;
		this.coinCategory = coinCategory;
		this.maxCoinCount = maxCoinCount;
		this.nowCoinCount = nowCoinCount;
	}
}

File: Machine/Models/ViewModel/CoinView.cs
using System;
using Machine.Models.Enum;

namespace Machine.Models.ViewModel
{
	public class CoinView
	{
		public Int32 Id { get; set; }
		public CoinCategory CoinCategory { get; set; }
		public Int32 MaxCoinCount { get; set; }
		public Int32 NowCoinCount { get; set; }

		public CoinView() { }

		public CoinView(Int32 id, CoinCategory coinCategory, Int32 maxCoinCount, Int32 nowCoinCount)
		{
			Id = id;
			CoinCategory = coinCategory;
			MaxCoinCount = maxCoinCount;
			NowCoinCount = nowCoinCount;
		}
	}
}